plugin_manager.py

```diff
@@ -156,7 +156,11 @@ def final_load(self):
             time = 60
         await asyncio.sleep(time)
         self.logger.debug("Writing to shelve...")
-        self.shelve.sync()
+        try:
+            self.shelve.sync()
+        except Exception as e:
+            err = f"{e}." if e else ""
+            self.logger.error(f"Error writing to shelve. {err}")
```

Comment:
Better to `self.logger.exception("Error writing to shelve. ", exc_info=True)` here.